pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,174 @@
+            assert isinstance(
+                quantile_pred, torch.Tensor
+            ), "Quantile prediction should be a tensor."  # noqa: E501
+
+            # capability:quantile_generation is a tag for metrics
+            # that override the default `to_quantiles` method from `Metric` base class.
+            if y_pred.ndim == 3 or (
+                object_pkg.get_class_tag("shape:adds_quantile_dimension")
+            ):
+                assert quantile_pred.shape == (
+                    y_pred.shape[0],
+                    y_pred.shape[1],
+                    len(quantiles),
+                ), f"`to_quantiles()`, fails for metric {metric}()."
+            elif y_pred.ndim == 2:
+                assert quantile_pred.shape == (
+                    y_pred.shape[0],
+                    y_pred.shape[1],
+                    1,
+                ), f"`to_quantiles()`, fails for metric {metric}()."
+
+            # testing composite metrics
+            composite = metric + metric
+            weighted = metric * 0.5
+
+            composite_result = composite(y_pred, y)
+            weighted_result = weighted(y_pred, y)
+
+            assert isinstance(composite_result, torch.Tensor)
+            assert isinstance(weighted_result, torch.Tensor)
+            assert torch.isfinite(composite_result).all()
+            assert torch.isfinite(weighted_result).all()
+
+    @pytest.mark.parametrize("target_type", ["standard", "packed", "weighted"])
+    @pytest.mark.parametrize("reduction", ["mean", "none", "sqrt-mean"])
+    def test_reduction_modes(
+        self, object_pkg, object_class, request, target_type, reduction
+    ):  # noqa: E501
+        """Test that all metrics support different reduction modes."""
+
+        prepared_data = self._setup_metric_test_scenario(
+            object_pkg, object_class, target_type, request
+        )
+
+        if prepared_data is None:
+            return None
+
+        metric, y_pred, y = prepared_data
+
+        if (
+            reduction == "sqrt-mean"
+            and object_pkg.get_class_tag("distribution_type") == "normal"
+        ):  # noqa: E501
+            return None  # sqrt-mean is not applicable for normal distribution
+
+        metric.update(y_pred, y)
+        result = metric.reduce_loss(metric.losses, metric.lengths, reduction=reduction)  # noqa: E501
+        assert isinstance(result, torch.Tensor)
+
+        if reduction == "none":
+            result.shape == metric.losses
+        else:
+            assert result.ndim == 0
+            if reduction == "sqrt-mean":
+                assert (
+                    result >= 0
+                ), "Result should be non-negative for sqrt-mean reduction."  # noqa: E501
+
+    @pytest.mark.parametrize("target_type", ["standard", "packed", "weighted"])
+    def test_metric_functionality(self, object_pkg, object_class, request, target_type):
+        """Test metric functionality with appropriate test data."""
```

Comment:
best practice: add a bit more detail what gets tested here. "functionality" is very non-descript